jaq-interpret/src/path.rs

```diff
@@ -59,7 +59,9 @@ impl Part<Vec<Val>> {
                     let until = rel_bounds(until).map(move |i| Ok(abs_bound(i?, len, len)));
                     Box::new(prod(from, until).map(move |(from, until)| {
                         let (skip, take) = skip_take(from?, until?);
-                        Ok(Val::str(s.chars().skip(skip).take(take).collect()))
+                        Ok(Val::str(
+                            s.chars().skip(skip).take(take).collect::<String>(),
+                        ))
```

Comment:
This doesn't seem to be related to formatters?

Reply:
Ah, that's because you modified the str function.
I think that accepting parameters implementing Into should be done quite sparingly, as it can require more type annotations, like here. May I ask you for your motivation for having changed the signature of the function?